callstack/org.eclipse.tracecompass.incubator.analysis.core/src/org/eclipse/tracecompass/incubator/analysis/core/reports/ReportsDataProviderFactory.java

```diff
@@ -0,0 +1,729 @@
+        }
+
+        IDataProviderDescriptor descriptor = provider.createDescriptor(trace, config);
+
+        fTmfConfigurationTable.put(config.getId(), trace.getUUID(), config);
+        fTmfConfigurationHierarchy.put(trace.getUUID(), config.getId(), (String) config.getParameters().get(PARENT_ID));
+
+        writeConfiguration(trace, config);
+
+        return descriptor;
+    }
+
+    /**
+     * Remove a data provider descriptor and all its children recursively
+     *
+     * @param trace
+     *            the trace
+     * @param descriptor
+     *            the descriptor to remove
+     * @throws TmfConfigurationException
+     *             if an error occurs while removing
+     */
+    @Override
+    public void removeDataProviderDescriptor(@NonNull ITmfTrace trace, @NonNull IDataProviderDescriptor descriptor) throws TmfConfigurationException {
+        ITmfConfiguration config = descriptor.getConfiguration();
+        if (config == null) {
+            throw new TmfConfigurationException("Data provider was not created by a configuration"); //$NON-NLS-1$
+        }
+
+        ReportProviderType type = getReportType(config);
+        if (type == ReportProviderType.NONE) {
+            // For NONE type, use this factory's removeDescriptor
+            removeDescriptor(trace, config);
+        } else {
+            // For other types, use their specific providers
+            IReportDataProvider provider = ReportsDataProviderRegistry.getProvider(type);
+            if (provider != null) {
+                provider.removeDescriptor(trace, config);
+            }
+
+            removeConfiguration(trace, config);
+
+            // Clean up tables
+            fTmfConfigurationTable.remove(config.getId(), trace.getUUID());
+            fTmfConfigurationHierarchy.remove(trace.getUUID(), config.getId());
+        }
+    }
+
+    /**
+     * Get the available configuration source types
+     *
+     * @return the configuration source types
+     */
+    @SuppressWarnings("null")
+    @Override
+    public @NonNull List<@NonNull ITmfConfigurationSourceType> getConfigurationSourceTypes() {
+        List<ITmfConfigurationSourceType> result = new ArrayList<>();
+
+        // Add base configuration source type with the common descriptors
+        List<ITmfConfigParamDescriptor> baseDescriptors = new ArrayList<>(CONFIGURATION_SOURCE_TYPE.getConfigParamDescriptors());
+        baseDescriptors.addAll(List.of(
+                new TmfConfigParamDescriptor.Builder()
+                        .setKeyName(REPORT_TYPE)
+                        .setDescription(NLS.bind(REPORT_TYPE_DESC, ProviderType.NONE.name().toUpperCase()))
+                        .setIsRequired(true)
+                        .build(),
+                new TmfConfigParamDescriptor.Builder()
+                        .setKeyName(PARENT_ID)
+                        .setDescription(PARENT_ID_DESC)
```

Comment:
So, the user has to provide a unique parentID, right? This parentID is the ID of the container created. You should mention in the description that it needs to be unique.